src/librustdoc/json/mod.rs

```diff
@@ -32,36 +32,35 @@ crate struct JsonRenderer<'tcx> {
     index: Rc<RefCell<FxHashMap<types::Id, types::Item>>>,
     /// The directory where the blob will be written to.
     out_path: PathBuf,
+    cache: Rc<Cache>,
 }
 
 impl JsonRenderer<'_> {
     fn sess(&self) -> &Session {
         self.tcx.sess
     }
 
-    fn get_trait_implementors(
-        &mut self,
-        id: rustc_span::def_id::DefId,
-        cache: &Cache,
-    ) -> Vec<types::Id> {
-        cache
+    fn get_trait_implementors(&mut self, id: rustc_span::def_id::DefId) -> Vec<types::Id> {
+        self.cache
+            .clone()
             .implementors
             .get(&id)
             .map(|implementors| {
                 implementors
                     .iter()
                     .map(|i| {
                         let item = &i.impl_item;
-                        self.item(item.clone(), cache).unwrap();
+                        self.item(item.clone()).unwrap();
                         item.def_id.into()
                     })
                     .collect()
             })
             .unwrap_or_default()
     }
 
-    fn get_impls(&mut self, id: rustc_span::def_id::DefId, cache: &Cache) -> Vec<types::Id> {
-        cache
+    fn get_impls(&mut self, id: rustc_span::def_id::DefId) -> Vec<types::Id> {
+        self.cache
+            .clone()
```

Comment:
Why is this cloned?

Reply:
Same as above.

Reply:
Hum my bad, it's uneeded here I think. Checking.

Reply:
No, it throws this error:

```
error[E0500]: closure requires unique access to `self` but it is already borrowed
  --> src/librustdoc/json/mod.rs:47:18
   |
44 |         self.cache
   |         ---------- borrow occurs here
...
47 |             .map(|implementors| {
   |              --- ^^^^^^^^^^^^^^ closure construction occurs here
   |              |
   |              first borrow later used by call
...
52 |                         self.item(item.clone()).unwrap();
   |                         ---- second borrow occurs due to use of `self` in closure
```

Reply:
Ok, I think you can work around that by making a separate variable after the call to `get()`. I really would prefer to avoid the clone here, cache is *giant*. If you really can't make it work, you could add an Rc instead.

Reply:
No, it's a `Rc`. :)